services/testspace/testspace-base.js

```diff
@@ -0,0 +1,61 @@
+'use strict'
+
+const Joi = require('joi')
+const { nonNegativeInteger } = require('../validators')
+const { BaseJsonService, NotFound } = require('..')
+
+// https://help.testspace.com/docs/reference/web-api#list-results
+// case_counts|array|The contained cases [passed, failed, na, errored]|counters of result
+// session_* fields are for manual runs
+// There are instances where the api returns a 200 status code with an empty array
+// notably in cases where a space id is used
+const schema = Joi.array()
+  .items(
+    Joi.object({
+      case_counts: Joi.array()
+        .items(nonNegativeInteger)
+        .min(4)
+        .max(4)
+        .required(),
+    })
+  )
+  .required()
+
+// https://help.testspace.com/docs/dashboard/overview-navigate
+// Org is owner/account
+// Project is generally a repository
+// Space is a container, often a branch
+module.exports = class TestspaceBase extends BaseJsonService {
+  static category = 'build'
+  static defaultBadgeData = { label: 'tests' }
+
+  async fetch({ org, project, space }) {
+    // https://help.testspace.com/docs/reference/web-api#list-results
+    const url = `https://${org}.testspace.com/api/projects/${encodeURIComponent(
+      project
```

Comment:
Project names by default seem to follow the `org-name:repo-name` pattern so think encoding here is necessary